### PR TITLE
Setup gradual process for migrating to Typescript

### PR DESCRIPTION
**Pull request by s1na**
It obviously won't be possible to migrate the whole library to typescript at once. This PR sets up a process for migrating gradually. In the meanwhile JS and TS code will have to co-exist. To account for that:

- Dropped `babel` dependencies
- Added a simple (not too strict) `tsconfig.json` which allows JS code. All code is transpiled via `tsc` (even JS)
- Consequently JS code can't import TS files directly. Therefore I've changed the API tests to use the compiled version (`dist`) rather than source. This is until all source code has been migrated and can be later on reverted

For the first round, I've migrated `Bloom`, `Stack` and `Memory` (easier picks :smile:):

- `Bloom`'s API has also been limited to only accept `Buffer`. Since this was only used internally, it shouldn't be a breaking change
- `Stack` also only accepts `BN` (dropped `Buffer` as it wasn't being used)
- Added `rlp` as a dev dependency (due to version mismatch in `ethereumjs-util`, will make a PR for that)

Please let me know if you think this gradual process is suitable or not.

(To be merged into #479)

### CODE REVIEW

File: lib/evm/memory.ts [hunk 1]
```diff
@@ -66,7 +66,7 @@ module.exports = class Memory {
   }
 }
 
-const ceil = (value, ceiling) => {
+const ceil = (value: number, ceiling: number): number => {
   const r = value % ceiling
   if (r === 0) {
     return value
```
Comment (holgerd77) [inline lib/evm/memory.ts]: OK (file).

File: lib/evm/stack.ts [hunk 2]
```diff
@@ -73,26 +80,12 @@ module.exports = class Stack {
...
-      }
-    } else if (Buffer.isBuffer(value)) {
-      if (value.length <= 32) {
-        return true
-      }
-    }
-
-    return false
-  }
 }
```
Comment (holgerd77) [inline lib/evm/stack.ts]: OK (file), yes, makes totally sense to inline this private method.

File: package.json [hunk 3]
```diff
@@ -35,14 +37,14 @@
     "VM"
   ],
   "dependencies": {
-    "@babel/runtime": "^7.4.0",
     "async": "^2.1.2",
     "async-eventemitter": "^0.2.2",
+    "core-js-pure": "^3.0.1",
```
Comment (holgerd77) [inline package.json]: Seems to be ok as a new dependency, see [here](https://github.com/zloirock/core-js).

File: tests/api/bloom.js [hunk 4]
```diff
@@ -22,42 +22,42 @@ tape('bloom', (t) => {
...
+    b1.add(utils.toBuffer('value 1'))
     const b2 = new Bloom()
-    b2.add('value 2')
+    b2.add(utils.toBuffer('value 2'))
 
     b1.or(b2)
-    st.true(b1.check('value 2'), 'should contain "value 2" after or')
+    st.true(b1.check(utils.toBuffer('value 2')), 'should contain "value 2" after or')
     st.end()
   })
```
Comment (holgerd77) [inline tests/api/bloom.js]: OK, test adoption to the stricter ``Bloom`` filter API.

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  "compilerOptions": {
+    "outDir": "./dist",
+    "allowJs": true,
+    "target": "es5",
+    "lib": ["es2018"],
+    "sourceMap": true,
+    "strict": true
+  },
+  "include": ["./lib/**/*"]
+}
```
Comment (holgerd77) [inline tsconfig.json]: Rest of the changes until here ok.

**Comment by holgerd77** [comment]
Side note 1: not mission-critical for the moment, just for the state of stuff: prettier/formatting is not working yet, or is it?

Side note 2: let me know what concretely you are working on throughout the day, eventually (possible not, hmm 😛) I will also do one or two file transitions if I find the time.

**Comment by s1na** [comment]
> prettier/formatting is not working yet, or is it?

Well, the command runs, but I haven't fixed the errors :-D They were mostly on the `md` files.

> Side note 2

Sure! Until Monday I'll be working very sporadically and can't really plan it, but from Monday we can divide and conquer together ;)

**Pull request merged**